utils/tests/test_unauthenticated_endpoints.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+import pytest
+from django_extensions.management.commands import show_urls
+import json
+import requests
+from lxml import html
+import mvj.urls
+
+
+def paths():
+    cmd = show_urls.Command()
+    views_str = cmd.handle(
+        no_color=True,
+        language=None, decorator=None,
+        format_style='pretty-json',
+        urlconf="ROOT_URLCONF",unsorted=True)
+    views = json.loads(views_str)
+    return [path for view in views if (path := view.get('url'))]
+
+repls = (('/<pk>\.<format>/','/1234.json/'),
+         ('/<pk>/','/1234/'),
+         ('/<path:object_id>/','/abc/123/'))
+
+loginpagetitle = 'Kirjaudu sisään | Django-sivuston ylläpito'
+
+whitelist = '''
+    /auth/login/
+    /auth/logout/
+    /v1/pub/answer/
+    /v1/pub/faq/
+    /v1/pub/form/
+    /v1/pub/plot_search_ui/0
+    /v1/pub/plot_search/
+    /v1/pub/plot_search_stage/
+    /v1/pub/plot_search_type/
+    /v1/pub/plot_search_subtype/
+    /v1/pub/plot_search_ui/1
+    /docs/
+    /redoc/
+''' .split()
+
+@pytest.mark.parametrize('path',paths())
+def test_urls(path):
+    if not path.endswith('/') or path in whitelist: pytest.skip()
+    for r in repls:
+        path = path.replace(*r)
+    if '>' in path: pytest.skip() # This includes lone '\.<format>' and few corner cases
+    url = f'http://localhost:8001{path}'
+    res = requests.get(url)
+    ct = res.headers.get('content-type')
+    sc = res.status_code
+    if sc == 200 :
+        assert ct.startswith('text/html;'), f"Got {ct} response # curl -L {url}"
```

Comment:
You would want to send requests with the header`Content-Type: application/json` so you would get responses as json, the html responses are not expected to work (they come from the Django Restframeworks utility of having a swagger like UI)

Reply:
Header added, although it seem to have no effect to the content type of the response.

Reply:
Did you commit and push the code? I don't see this change.
If you're having doubts, lets setup a call to see what the issue is

Reply:
My bad, it obv should be the accept header not content-type :see_no_evil: 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept